lib/src/server/handler.dart

```diff
@@ -70,6 +70,10 @@ class ServerHandler extends ServiceCall {
   /// Emits a ping everytime data is received
   final Sink<void>? onDataReceived;
 
+  final StreamController<void> _onCanceledController = StreamController();
```

Comment:
Consider instead:

```
final Completer<void> _isCanceledCompleter = Completer<void>();

Future<void> onCanceled => _isCanceledCompleter.future;

bool get isCanceled => _isCanceledCompleter.isCompleted;
set isCanceled(bool value) {
  if (!isCanceled) {
    _isCanceledCompleter.complete();
  }
}
```

This way you can get read of the boolean. It also makes more sense to have a `Future` vs `Stream` here because it is a single shot event.